### PR TITLE
ant pattern

### PR DESCRIPTION
**Pull request by gailazar300**
- [X] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [X] This pull request is on the dev branch.
- [X] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Comment by gailazar300** [comment]
I have read the CLA Document and I hereby sign the CLA

File: artifactory/cli.go [hunk 1]
```diff
@@ -1938,6 +1938,9 @@ func uploadCmd(c *cli.Context) error {
 	if c.NArg() > 0 && c.IsSet("spec") {
 		return cliutils.PrintHelpAndReturnError("No arguments should be sent when the spec option is used.", c)
 	}
+	if c.IsSet("regexp") && c.IsSet("ant") {
```
Comment (eyalbe4) [inline artifactory/cli.go:1941]: This check should be added into the ```spec.ValidateSpec``` function, Notice that this function is called a few lines below. Letting this function handle the validation, takes care fo both the flags and the file-spec validation.
Something else to notice:
The following flag combinations are valid:
```
--ant --regexp=false
--ant=false --regexp
--ant=treu --regexp=fase
```
and so on.
The ```spec.ValidateSpec``` function takes care of all of that very easilly.

File: utils/cliutils/commandsflags.go [hunk 2]
```diff
@@ -734,6 +741,10 @@ var flagsMap = map[string]cli.Flag{
 		Name:  regexpFlag,
 		Usage: "[Default: false] Set to true to use a regular expression instead of wildcards expression to collect files to be added to the build info.` `",
 	},
+	badAnt: cli.BoolFlag{
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:744]: It looks like this flag hasn't been added to the ```BuildAddDependencies``` slice, which includes all the flags of the ```build-add-dependencies``` command.

File: artifactory_test.go [hunk 3]
```diff
@@ -4461,3 +4465,39 @@ func uploadWithSpecificServerAndVerify(t *testing.T, cli *tests.JfrogCli, server
 	assert.Len(t, searchItemsInArtifactory(t, tests.SearchRepo1ByInSuffix), expectedResults)
 	return nil
 }
+
+//This test checks the case of using --ant and --regexp at the same time
```
Comment (eyalbe4) [inline artifactory_test.go:4469]: //This --> // This
(Our conversion is to add one white-space after //).

File: go.mod [hunk 4]
```diff
@@ -19,9 +19,9 @@ require (
 	gopkg.in/yaml.v2 v2.3.0
 )
 
-// replace github.com/jfrog/jfrog-client-go => github.com/jfrog/jfrog-client-go dev
+replace github.com/jfrog/jfrog-client-go => ../jfrog-client-go
```
Comment (eyalbe4) [inline go.mod:22]: To be reverted before committing.

File: go.mod [hunk 5]
```diff
@@ -19,9 +19,9 @@ require (
...
 
-// replace github.com/jfrog/jfrog-cli-core => github.com/jfrog/jfrog-cli-core dev
+replace github.com/jfrog/jfrog-cli-core => ../jfrog-cli-core
```
Comment (eyalbe4) [inline go.mod:24]: To be reverted before committing.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Review by eyalbe4 — APPROVED** [review]

**Pull request merged**